orion/controllers/programs/programsController.go

```diff
@@ -81,6 +79,40 @@ func DeleteProgram(c *gin.Context) {
   return
 }
 
-func CheckValidProgram() bool {
+func CheckValidProgram(c *gin.Context) bool {
+  // Retrieves the inputted values
+  name := c.Param("programName")
+  grade1 := c.Param("grade1")
+  grade2 := c.Param("grade2")
+
+  // Checks if program name is empty
+  if name == "" {
+    c.String(http.StatusBadRequest, "Invalid name " + name)
+    return false
+  }
+
+  // Checks if the program name is a string only
+  for _, i := range name {
+    if (i < 'a' || i > 'z') && (i < 'A' || i > 'Z') && i != '_' {
+      c.String(http.StatusBadRequest, "Invalid name " + name)
```

Comment:
since we're not using gin, you won't need these c.String(...). I'll tell you later what these are for.